vllm/core/block_manager_v2.py

```diff
@@ -100,6 +100,9 @@ def __init__(
         self.block_tables: Dict[SeqId, BlockTable] = {}
         self.cross_block_tables: Dict[EncoderSeqId, BlockTable] = {}
 
+        self._cached_computed_seq_blocks: Dict[SeqId, List[int]] = {}
+        self._seq_last_access: Dict[SeqId, float] = {}
```

Comment:
There was a significant overhead in these function calls, since they traversed the full block lists.

Reply:
Can we modify the API such that it allows caching the result // we don't have to traverse the full block lists? 

Two downsides:
* we expose more complexity in this layer than is necessary (this is tech debt we can live with, if it's too hard)
* we make it harder for other block managers to use prefix caching (we may have a block manager which specializes for another type, e.g. the newer models which use sliding window + normal attention).

Reply:
This is a good idea. I have refactored this logic out to two classes: ComputedBlocksTracker and LastAccessBlocksTracker so it will be easier to port the logic to other places.